dartagnan/src/test/java/com/dat3m/dartagnan/utils/rules/Providers.java

```diff
@@ -60,11 +63,13 @@ public static Provider<Program> createProgramFromFile(Supplier<File> fileSupplie
     // =========================== Task related providers ==============================
 
     public static Provider<VerificationTask> createTask(Supplier<Program> programSupplier, Supplier<Wmm> wmmSupplier, Supplier<EnumSet<Property>> propertySupplier,
-                                                        Supplier<Arch> targetSupplier, Supplier<Integer> boundSupplier, Supplier<Integer> timeoutSupplier) {
+                                                        Supplier<Arch> targetSupplier, Supplier<Integer> boundSupplier, boolean initiRegs) {
     	return Provider.fromSupplier(() -> VerificationTask.builder().
+    	        withConfig(Configuration.builder()
+    	                .setOption(INITIALIZE_REGISTERS, String.valueOf(initiRegs))
+    	                .build()).
     			withTarget(targetSupplier.get()).
     			withBound(boundSupplier.get()).
-    			withSolverTimeout(timeoutSupplier.get()).
     			build(programSupplier.get(), wmmSupplier.get(), propertySupplier.get()));
     }
```

Comment:
I would rather have this method take a `Provider<Configuration>` as input and then create the appropriate provider in the unit tests. It's a lot more general.